spec/features/viewing_relationship_page_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'rails_helper'
+
+describe "Relationship Page", :type => :feature do
+  before(:all) { @user = create_user_with_sf }
+  let(:org) { create(:org) }
+  let(:person) { create(:org) }
+  let(:relationship) do
+    rel = Relationship.create!(category_id: 12, entity: org, related: person, last_user_id: @user.sf_guard_user.id)
+    Reference.create!(object_id: rel.id, object_model: "Relationship", source: "https://example.com")
+    rel
+  end
+
+  context "Anonymous user" do
+    before(:each) do
+      visit "/relationships/#{relationship.id}"
+    end
+
+    it 'has relationship title' do
+      expect(page).to have_selector 'h1.relationship-title-link a', text: relationship.name
+    end
+
+    context 'source links' do
+      it 'has one source link' do
+        expect(page).to have_selector '#source-links-table tbody tr', count: 1, text: "https://example.com"
+      end
+
+      it 'opens the source links in a new tab' do
+        page.all('#source-links-table tbody tr a').each do |element|
+          expect(element[:target]).to eql '_blank'
```

Comment:
as a matter of curiosity (this test is fine!): i expected this test to click on one link and assert that the url of the current page had changed. what was the reasoning for writing it this way? faster test? capybara not up and running? wanting to test *all* links?

Reply:
I don't want the url of the page to change - -clicking on the link should open the link in a new tab.

This isn't being run in a 'real' browser or even a mock browser, it's using [RackTest](https://github.com/teamcapybara/capybara#racktest). I don't think that it's capable of opening external links, and it certainly won't have tabs.  if we get selenium working, we could test if a new tab is opened.